app.go

```diff
@@ -116,7 +128,34 @@ func (io invokeOption) String() string {
 	return fmt.Sprintf("fx.Invoke(%s)", strings.Join(items, ", "))
 }
 
-// Options composes a collection of Options into a single Option.
+// Options composes a collection of Options into a single Option. This allows
+// packages to bundle sophisticated functionality into easy-to-use Fx modules.
+// For example, a logging package might export a simple option like this:
+//
+//  package logging
+//
+//  var Module = fx.Provide(func() *log.Logger {
+//    return log.New(os.Stdout, "", 0)
+//  })
+//
+// A shared all-in-one microservice package could then use Options to bundle
```

Comment:
@abhinav just wrote a section on how not to do this :(

Is there a better example we can use for composition that doesn't cross library bounds?

Reply:
Doing this is basically the whole point - that's what `uberfx` is internally. Why not mention this?

Reply:
As always, this is nuanced - the pattern referenced here is fine, as long as it preserves the user's ability to break glass and consume each module separately. Abhinav's point is that some compositions of options make this quite a bit harder.

Rather than rewording this section, which is IMO the strongest selling point of Fx overall, I've added a more general "be cautious" note to the end.